solidity/test/AcreMultiAssetVault.test.ts

```diff
@@ -0,0 +1,949 @@
+              expectedDepositId,
+              depositAmount,
+            )
+        })
+
+        it("should transfer the asset to the Mezo Portal", async () => {
+          await expect(tx).to.changeTokenBalances(
+            asset1,
+            [caller, multiAssetVault, mezoPortal],
+            [-depositAmount, 0, depositAmount],
+          )
+        })
+
+        it("should store deposit data", async () => {
+          const depositData = await multiAssetVault.getDeposit(
+            await depositOwner.getAddress(),
+            await asset1.getAddress(),
+            expectedDepositId,
+          )
+
+          expect(depositData.balance, "invalid balance").to.equal(depositAmount)
+          expect(depositData.mezoDepositId, "invalid mezo deposit id").to.equal(
+            initialMezoDepositCount + 1n,
+          )
+        })
+      })
+
+      describe("when the deposit owner is different from the caller", () => {
+        beforeAfterSnapshotWrapper()
+
+        let caller: HardhatEthersSigner
+        let depositOwner: HardhatEthersSigner
+        let depositAmount: bigint
+
+        let initialDepositCount: bigint
+        let expectedDepositId: bigint
+
+        let callResult: bigint
+        let tx: ContractTransactionResponse
+
+        before(async () => {
+          caller = depositor1
+          depositOwner = depositor2
+          depositAmount = to1e18(10)
+
+          initialDepositCount = await multiAssetVault.depositCount()
+          expectedDepositId = initialDepositCount + 1n
+
+          await asset1.mint(await caller.getAddress(), depositAmount)
+          await asset1
+            .connect(caller)
+            .approve(await multiAssetVault.getAddress(), depositAmount)
+
+          callResult = await multiAssetVault
+            .connect(caller)
+            .depositFor.staticCall(
+              await asset1.getAddress(),
+              depositAmount,
+              await depositOwner.getAddress(),
+            )
+          tx = await multiAssetVault
+            .connect(caller)
+            .depositFor(
+              await asset1.getAddress(),
+              depositAmount,
+              await depositOwner.getAddress(),
+            )
+        })
+
+        it("should return the deposit id", () => {
+          expect(callResult).to.be.equal(expectedDepositId)
+        })
+
+        it("should increment deposit count", async () => {
+          expect(await multiAssetVault.depositCount()).to.be.equal(
+            initialDepositCount + 1n,
+          )
+        })
+
+        it("should emit DepositCreated event", async () => {
+          await expect(tx)
+            .to.emit(multiAssetVault, "DepositCreated")
+            .withArgs(
+              await depositOwner.getAddress(),
+              await asset1.getAddress(),
+              expectedDepositId,
+              depositAmount,
+            )
+        })
+
+        it("should transfer the asset to the vault", async () => {
```

Comment:
To the Mezo Portal.

Reply:
6bd6ff4df9709343674e7f6c96c2283b7fbcc80d